src/handlers/issue-deduplication.ts

```diff
@@ -0,0 +1,138 @@
+import { IssueSimilaritySearchResult } from "../adapters/supabase/helpers/issues";
+import { Context } from "../types";
+const MATCH_THRESHOLD = 0.95;
+const WARNING_THRESHOLD = 0.5;
+
+export interface IssueGraphqlResponse {
+  node: {
+    title: string;
+    url: string;
+  };
+}
+
+/**
+ * Check if an issue is similar to any existing issues in the database
+ * @param context
+ * @returns true if the issue is similar to an existing issue, false otherwise
+ */
+export async function issueChecker(context: Context): Promise<boolean> {
+  const {
+    logger,
+    payload,
+    adapters: { supabase },
+  } = context;
+
+  const issue = payload.issue;
+
+  //First Check if an issue with more than MATCH_THRESHOLD similarity exists (Very Similar)
+  const similarIssue = await supabase.issue.findSimilarIssues(issue.body + issue.title, MATCH_THRESHOLD, issue.node_id);
+  if (similarIssue && similarIssue?.length > 0) {
+    logger.info(`Similar issue which matches more than ${MATCH_THRESHOLD} already exists`);
+    //Close the issue as "unplanned"
+    await context.octokit.issues.update({
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      issue_number: issue.number,
+      state: "closed",
+      labels: ["unplanned"],
```

Comment:
No labels

Reply:
Removed Labels. Labels will not be added on issue close